eZ/Publish/Core/Search/Common/DocumentMapper.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Search\Common;
+
+use eZ\Publish\SPI\Persistence\Content;
+use eZ\Publish\SPI\Persistence\Content\Location;
+use eZ\Publish\SPI\Persistence\Content\Handler as ContentHandler;
+use eZ\Publish\SPI\Persistence\Content\Location\Handler as LocationHandler;
+use eZ\Publish\SPI\Persistence\Content\Type\Handler as ContentTypeHandler;
+use eZ\Publish\SPI\Persistence\Content\ObjectState\Handler as ObjectStateHandler;
+use eZ\Publish\SPI\Persistence\Content\Section\Handler as SectionHandler;
+
+/**
+ * Mapper maps Content and Location objects to a Document objects, representing a
+ * document in search index storage.
+ *
+ * Note that custom implementations might need to be accompanied by custom schema.
+ */
+abstract class DocumentMapper
```

Comment:
That measn the change from `class StandardMapper implements MapperInterface` is not needed either.

@alongosz there might be things in common across all three still, but then just use an optional trait.